pkg/controllers/types.go

```diff
@@ -46,7 +46,7 @@ type Controller interface {
 	//   a. Source: the resource that is being watched
 	//   b. EventHandler: which controller objects to be reconciled
 	//   c. WatchesOption: which events can be filtered out before processed
-	Watches(context.Context) (source.Source, handler.EventHandler, builder.WatchesOption)
```

Comment:
We can remove the Interval() API here now as well, right?

Reply:
I'm keeping the Interval() here as the allocation an reallocation controllers still rely on it.